pylint/utils/docs.py

```diff
@@ -42,27 +42,29 @@ def _get_checkers_infos(linter: PyLinter) -> dict[str, dict[str, Any]]:
     return by_checker
 
 
-def _get_checkers_documentation(linter: PyLinter) -> str:
+def _get_checkers_documentation(linter: PyLinter, show_options: bool = True) -> str:
     """Get documentation for individual checkers."""
-    result = get_rst_title("Pylint global options and switches", "-")
-    result += """
+    result = ""
+    if show_options:
+        result += get_rst_title("Pylint global options and switches", "-")
+        result += """
 Pylint provides global options and switches.
 
 """
-    for checker in linter.get_checkers():
-        name = checker.name
-        if name == MAIN_CHECKER_NAME:
-            if checker.options:
-                with warnings.catch_warnings():
-                    warnings.filterwarnings("ignore", category=DeprecationWarning)
-                    for section, options in checker.options_by_section():
-                        if section is None:
-                            title = "General options"
-                        else:
-                            title = f"{section.capitalize()} options"
-                        result += get_rst_title(title, "~")
-                        assert isinstance(options, list)
-                        result += f"{get_rst_section(None, options)}\n"
+        for checker in linter.get_checkers():
+            name = checker.name
+            if name == MAIN_CHECKER_NAME:
+                if checker.options:
+                    with warnings.catch_warnings():
+                        warnings.filterwarnings("ignore", category=DeprecationWarning)
+                        for section, options in checker.options_by_section():
```

Comment:
``options_by_section`` is deprecated, it might be a good thing to find a solution without using it so the transition to 3.0 is easier. Is it a problem ?

Reply:
I don't see this as a priority. We could also make it a private function in `3.0` and keep using it ourselves. The important thing is that downstream packages don't depend on it.

Reply:
Right, that make sense.